internal/pkg/logger/log.go

```diff
@@ -1,25 +1,108 @@
 package log
 
 import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/joshsoftware/peerly-backend/internal/pkg/constants"
 	l "github.com/sirupsen/logrus"
+	"gopkg.in/natefinch/lumberjack.v2"
 )
 
+var Logger *l.Logger
+
+var lumberjackLogger = &lumberjack.Logger{
+	MaxSize:    100, // megabytes
+	MaxBackups: 3,   // number of log files
+	MaxAge:     365, // days
+	Compress:   true,
+}
+
 // Error - prints out an error
-func Error(appError error, msg string, triggeringError error) {
-	l.WithFields(l.Fields{"appError": appError, "message": msg}).Error(triggeringError)
+func Error(ctx context.Context, args ...interface{}) {
+	requestID, ok := ctx.Value(constants.RequestID).(string)
+	if !ok {
+		requestID = ""
```

Comment:
instead of empty use "N/A"

Reply:
created new function and if requestID is empty "N/A" used